src/edu/ucsb/cs56/projects/music/beatbox/BeatBoxFinal.java

```diff
@@ -180,12 +211,46 @@ public void buildTrackAndStart() {
 	    sequencer.start() ;
 	    sequencer.setTempoInBPM(120) ;
 	} catch(Exception e) {e.printStackTrace() ;}
+        for(int i = 0; i < 16; i++){
+            for(int j = 0; j < 16; j++){
+                JCheckBox box = (JCheckBox) checkboxList.get(j + (i*16));
+                box.addActionListener(more);
+            }
+        }   // add a way to update sound
```

Comment:
are you _sure_ this is the right place to put this code?

Reply:
after reading through the code I understand what you're trying to do - you're restarting the sequencer every time you click a box, so you need to make sure you don't restart the sequencer while it's not playing. But wouldn't it be _so_ much simpler to just check if the sequencer was stopped, and if so, not start it, instead of constantly adding and removing 256 different listeners every time the music starts or stops?